shopinvader/services/cart.py

```diff
@@ -149,21 +150,64 @@ def _add_item(self, cart, params):
         else:
             with self.env.norecompute():
                 vals = self._prepare_cart_item(params, cart)
-                # the next statement is done with suspending the security for
-                #  performance reasons. It is safe only if both 3 following
-                # fields are filled on the sale order:
-                # - company_id
-                # - fiscal_position_id
-                # - pricelist_id
-                new_values = (
-                    self.env["sale.order.line"]
-                    .suspend_security()
-                    .play_onchanges(vals, vals.keys())
-                )
+                new_values = self._sale_order_line_onchange(vals)
                 vals.update(new_values)
-                self.env["sale.order.line"].create(vals)
+                self._create_sale_order_line(vals)
             cart.recompute()
 
+    @contextmanager
+    def _simulate_anonymous_lang(self, values):
```

Comment:
@acsonefho `s/simulate_anonymous_lang/_force_ctx_lang/` or something else. IMO it's not a simulation, we wan't to force the usage of the lang specified into the context....